doc/CMakeLists.txt

```diff
@@ -21,8 +21,17 @@ if( EXISTS ${XELATEX} )
   # add a configure rule to the header file though
   configure_file(${PROJECT_SOURCE_DIR}/title.tex.in ${PROJECT_SOURCE_DIR}/title.tex)
 
-  # Add custom target to build only documentation
-  add_custom_target(docs ALL)
+  # Add custom target to build only documentation, this target will depend on all individual PDFs
+  get_directory_property(hasParent PARENT_DIRECTORY)
+  if(hasParent AND BUILD_DOCS_ONLY_WITH_PACKAGE)
+    # Because we don't want to rebuild doc automatically, we do not add it to the 'ALL' rule
+    message(AUTHOR_WARNING "The target `docs` was added, but not to the default build rule. It will be triggered by `make docs` or `make package`")
+    add_custom_target(docs)
+  else()
+    # Building standalone, so add to ALL
+    message(AUTHOR_WARNING "The target `docs` was added to the default build rule")
```

Comment:
I would have made `BUILD_DOCS_ONLY_WITH_PACKAGE=ON` by default myself.
And therefore, since it isn't, both making it advanced and not displaying a message for like one iteration cycle at least is likely going to mean that many developpers might not notice this option at all.

That being said, if there is consensus that this is better to make it advanced and not display a message I'm totally fine with it and it's a very quick change.


----

As to your comment, above (there's no reply box):

 > This seems pretty hacky. I would think there would be a way to add a dependency to the install target, but apparently not. If it works for you then I don't see a problem with it.

I would have thought there would be a way too, but quickly realized there wasn't.

Reply:
I'd like to see these warnings go away.  At least the one for the default case.  These are going to pop up in my console every time I configure with BUILD_DOCS on.

My real preference would be to eliminate the `BUILD_DOCS` and `BUILD_DOCS_ONLY_WITH_PACKAGE` variables, and have one single variable that defines the documentation build config, with options instead of just a boolean status: `DOCUMENTATION_BUILD: [ DoNotBuild | BuildWithAll | BuildOnlyWithPackage ]` or something like that.  But I think that has too many possible ramifications.  

I'm OK with leaving things as they are, but not throwing a warning.